apps/vaporgui/ContourEventRouter.cpp

```diff
@@ -32,9 +34,26 @@ ContourEventRouter::ContourEventRouter(QWidget *parent, ControlExec *ce) : Rende
             _minValueSlider = new PDoubleSliderEditHLI<CP>("Minimum Value", &CP::GetContourMin, &CP::SetContourMin),
             new PConstantColorWidget
         }),
+        (new PShowIf(""))->DimensionEquals(3)->Then(new PGroup({
+            new PSection("Slice", {
+                (new PDoubleSliderEdit(RenderParams::SampleRateTag, "N Samples"))->SetRange(32, 2000)
+            })
+        })),
+    }));
+    
+    AddGeometrySubtab(new PGroup({
+        new PGeometrySubtab,
+        (new PShowIf(""))->DimensionEquals(3)->Then(new PGroup({
+            new PSection("Slice Rotation", {
+                (new PDoubleSliderEdit( RenderParams::XSlicePlaneRotationTag, "X"))->SetRange(-90.,90.)->EnableDynamicUpdate(),
+                (new PDoubleSliderEdit( RenderParams::YSlicePlaneRotationTag, "Y"))->SetRange(-90.,90.)->EnableDynamicUpdate(),
+                (new PDoubleSliderEdit( RenderParams::ZSlicePlaneRotationTag, "Z"))->SetRange(-90.,90.)->EnableDynamicUpdate(),
+            }),
+            new PSliceOriginSelector,
```

Comment:
The origin controls seem to be no-ops.  Should they be removed?

Reply:
I don't see how they are no-ops?

Reply:
They're not no-ops, but they can appear that way in some instances. @sgpearse see my comments [below](https://github.com/NCAR/VAPOR/pull/2992#issuecomment-1032865304).